oci/table_oci_queue_queue.go

```diff
@@ -0,0 +1,332 @@
+		return nil, err
+	}
+
+	// Build request parameters
+	request, isValid := buildQueueFilters(equalQuals, logger)
+	if !isValid {
+		return nil, nil
+	}
+	request.CompartmentId = types.String(compartment)
+	request.Limit = types.Int(1000)
+	request.RequestMetadata = oci_common.RequestMetadata{
+		RetryPolicy: getDefaultRetryPolicy(d.Connection),
+	}
+
+	limit := d.QueryContext.Limit
+	if limit != nil {
+		if *limit < int64(*request.Limit) {
+			request.Limit = types.Int(int(*limit))
+		}
+	}
+
+	pagesLeft := true
+	for pagesLeft {
+		response, err := session.QueueAdminClient.ListQueues(ctx, request)
+		if err != nil {
+                         logger.Error("oci_queue_queue. listQueues", "api_error", err)
+			return nil, err
+		}
+
+		for _, queueSummary := range response.Items {
+			d.StreamListItem(ctx, queueSummary)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+		if response.OpcNextPage != nil {
+			request.Page = response.OpcNextPage
+		} else {
+			pagesLeft = false
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getQueue(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getQueue")
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("getQueue", "Compartment", compartment, "OCI_REGION", region)
+
+	var id string
+	if h.Item != nil {
+		id = *h.Item.(queue.QueueSummary).Id
+	} else {
+		id = d.KeyColumnQuals["id"].GetStringValue()
+		// Restrict the api call to only root compartment/ per region
+		if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+			return nil, nil
+		}
+	}
+
+	if id == "" {
+		return nil, nil
+	}
+	// Create Session
+	session, err := queueService(ctx, d, region)
+	if err != nil {
+                logger.Error("oci_queue_queue. getQueue", "connection_error", err)
+		return nil, err
+	}
+
+	request := queue.GetQueueRequest{
+		QueueId: types.String(id),
+		RequestMetadata: oci_common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(d.Connection),
+		},
+	}
+	response, err := session.QueueAdminClient.GetQueue(ctx, request)
+	if err != nil {
+                logger.Error("oci_queue_queue. getQueue", "api_error", err)
+		return nil, err
+	}
+
+	return response.Queue, nil
+}
+
+// Build additional filters
+func buildQueueFilters(equalQuals plugin.KeyColumnEqualsQualMap, logger hclog.Logger) (queue.ListQueuesRequest, bool) {
+	request := queue.ListQueuesRequest{}
+	isValid := true
+
+	if equalQuals["displayName"] != nil && strings.Trim(equalQuals["displayName"].GetStringValue(), " ") != "" {
+		request.DisplayName = types.String(equalQuals["displayName"].GetStringValue())
+	}
+	return request, isValid
+}
+
+// Priority order for tags
+// 1. System Tags
```

Comment:
```suggestion
```

Safe to remove system tags from the list since they are not defined in `queueTags` function.